### PR TITLE
Adding in assertions and updating documentation as per review.

### PR DESCRIPTION
**Pull request by alanaharris22**

### CODE REVIEW

File: R/date_cut.R [hunk 1]
```diff
@@ -58,6 +60,15 @@ date_cut <- function(dataset_sdtm,
   assert_data_frame(dataset_cut,
     required_vars = vars(USUBJID, !!cut_var)
   )
+  assert_that(
+    (length(get_duplicates(dataset_cut$USUBJID)) == 0),
+    msg = "Duplicate patients in the DCUT (dataset_cut) dataset, please update."
+  )
+  assert_that(
+    (any(is.na(dataset_cut$DCUTDTM)) == FALSE),
```
Comment (barnett11) [inline R/date_cut.R:68]: Should we be using cut_var rather than DCUTDTM explicitly?

**Review by barnett11 — COMMENTED** [review]
Were you going to update the documentation also as per #153?

**Comment by alanaharris22** [comment]
Hi @barnett11 - yes the documentation has been updated already (see screenshot). Good point regarding DCUTDTM/cut_var - I've updated the non-missing cut date checks to use cut_var in date_cut() and special_dm_cut. I've actually removed the check from pt_cut since I'm not sure it makes sense there as we don't actually use this variable at all. 
image

File: R/date_cut.R [hunk 2]
```diff
@@ -58,6 +60,15 @@ date_cut <- function(dataset_sdtm,
   assert_data_frame(dataset_cut,
     required_vars = vars(USUBJID, !!cut_var)
   )
+  assert_that(
+    (length(get_duplicates(dataset_cut$USUBJID)) == 0),
+    msg = "Duplicate patients in the DCUT (dataset_cut) dataset, please update."
+  )
+  assert_that(
+    (any(is.na(mutate(dataset_cut, !!cut_var))) == FALSE),
+    msg = "At least one patient with missing DCUTDTM in the DCUT
```
Comment (barnett11) [inline R/date_cut.R:69]:
```suggestion
    msg = "At least one patient with missing datacut date (cut_var) in the DCUT
```
Suggest updating DCUTDTM reference in warning also

**Pull request merged**